app/models/sources/joyent.rb

```diff
@@ -26,6 +26,24 @@ def validate_config!
       return false
     end
 
+    # Connnect to Joyent and find all of the machines with that combination of
+    # user and datacenter that is provided with each source object source.
+    # Each machine found will be delete.
+    def purge_servers(captcha=nil)
+      if captcha.nil?
+        puts("What is 1 + 2 + 3 + 4? Put answer as a parameter.")
+      elsif captcha == 10
+        connection.servers.select do |s|
```

Comment:
That's a good point: the original requirements called for something that could basically say "get rid of all my Joyent proxies" or something similar, but we often have multiple `Source` objects that correspond to a single account. We probably shouldn't have a `Source` method that deletes another one's proxies; that can cause all sorts of weird invalid states.

My suggestion: the purge method should:

1. instruct each `Site` to forget each of its active proxies, then
2. destroy all of its proxies, then
3. destroy all of its connection's servers that it created (using `server_is_proxy_type?`).

Something like the following:
```ruby
# (not sure of all of this is strictly correct, but it should get you started)
self.proxies.active.each do |proxy|
  Site.all.each do |site|
    site.forget_proxy(proxy)
  end
end

self.proxies.each do |proxy|
  proxy.destroy
end

connection.servers.each do |server|
  if server_is_proxy_type? server
    server.destroy
  end
end
```

Some notes:

  * This may result in several requests to decommission each instance, but that shouldn't be too much of a problem unless `#destroy` raises an exception.
  * You'll need to make `Site#forget_proxy` public.